### PR TITLE
Generate MacOS framework with GN + Ninja.

### PR DESCRIPTION
**Pull request by godofredoc**
This is to separate the mac builds from tests and archives.

Bug: https://github.com/flutter/flutter/issues/81855

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by dnfield — APPROVED** [review]
LGTM

File: shell/platform/darwin/macos/BUILD.gn [hunk 1]
```diff
@@ -317,3 +321,32 @@ if (build_glfw_shell) {
     public_configs = [ "//flutter:config" ]
   }
 }
+
+zip_bundle("_zip_macos_flutter_framework") {
```
Comment (zanderso) [inline shell/platform/darwin/macos/BUILD.gn:325]: If the '_' is meant to communicate something about the visibility of this target, then that would be better accomplished by adding `visibility = [ ":*" ]`.

Comment (godofredoc) [reply shell/platform/darwin/macos/BUILD.gn]: Done! updated the zip_bundle template to include the visibility.

File: build/zip_bundle.gni [hunk 2]
```diff
@@ -42,6 +42,7 @@ template("zip_bundle") {
     script = "//flutter/build/zip.py"
     outputs = [ "$root_build_dir/zip_archives/${invoker.output}" ]
     sources = []
+    forward_variables_from(invoker, [ "visibility" ])
```
Comment (zanderso) [inline build/zip_bundle.gni:45]: optional nit: the forwards usually come first, and it looks like you could also forward `deps`.

**Review by zanderso — APPROVED** [review]

File: build/zip_bundle.gni (hunk 2 above)
Comment (godofredoc) [reply build/zip_bundle.gni:45]: done!

**Pull request merged**